File: FirstBadVersion.cpp
class Solution {
public:
    int firstBadVersion(int n) {
        long hi=n,lo=1;
        long mid,ans;
        
        while(hi>=lo){
            mid=(hi+lo)/2;
        if(isBadVersion(mid) == true){
            ans = mid;
            hi = mid-1;
        }else
            lo = mid+1;
            }
        return ans;
    }
};

File: FirstBadVersion.py
# Day :1
# https://leetcode.com/explore/challenge/card/may-leetcoding-challenge/534/week-1-may-1st-may-7th/3316/

class Solution:
    def firstBadVersion(self, n):
        start=0
        final=n
        while start <= final :
            mid=  start + ((final-start)//2)
            
            if isBadVersion(mid) ^ isBadVersion(mid-1) : return mid
            
            elif isBadVersion(mid): final =mid-1
            
            else: start =mid+1
        return -1
